calculator_test.go

```diff
@@ -2,23 +2,332 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
+type testCase struct {
```

Comment:
We usually don't bother to name the test case struct, because it tends to be unique to each test function, so we don't need to name it.

Reply:
@bitfield can you clarify what you mean by not naming it? Is there a way for me to not have to create a type testCase struct but use that struct type in my test cases?

Reply:
Yes, indeed! We can use what's called an _anonymous struct type_. For example:

```go
tcs := []struct{
	a, b float64
	want float64
}{
	{ a: 1, b: 2, want: 3},
	...
}
```

Reply:
@bitfield today i learned! Can't believe I haven't come across this concept before. I was attempting this without having to refer to your examples in the book, so I probably overlooked it there, but I honestly can't say I've seen it mentioned in other resources I've used. Very cool :D

Reply:
Yes, it's really common in tests, at least.